### PR TITLE
Simplified the creation of ArgumentsAreDifferent-Exceptions.

### LINKED ISSUE #426: Refactor instance based utility classes to static utility classes
The internal implementation of Mockito contains numerous helper classed that serve as a container for useful methods. By its nature these classes don't have a state. Currently some them are instance based in other word you need to create the utility class to call a helper method. This not only pollutes the heap but also the code cause instance methods can't be imported statically. 

Here is an example:

```java
Helper helper = new Helper();
if (helper.isInputValid(input)){
  [...] 
}
```

vs. static import of Helper.isInputValid

```java
if (isInputValid(input)){
  [...] 
}
```

The aim of this ticket is to identify canidates that can be refactored to static utility classes. If you like to refactoring and mockito feel free to send a PR and reference this issue.

Refactoring canidates:
- [ ] `AccessibilityChanger`
- [ ] `BeanPropertySetter`
- [ ] `ConditionalStackTraceFilter`
- [ ] `FieldCopier`
- [ ] `FieldReader`
- [ ] `GenericMaster` should be integrate into `GenericTypeResolver`
- [ ] `JUnitFailureHacker` can be removed when the deprecated `VerboseMockitoJUnitRunner` is removed
- [ ] `LenientCopyTool`
- [ ] `MatcherBinder`
- [ ] `MockitoCore` should better be a singleton
- [ ] `MockCreationValidator`
- [ ] `RemoveFirstLine`
- [x] #591 `ArgumentMatchingTool`
- [x] #515 `AllInvocationsFinder`
- [x] #502 `ArgumentsComparator`
- [x] #540 `ArrayUtils`
- [x] #490 `AtLeastXNumberOfInvocationsChecker`
- [x] #490 `AtLeastXNumberOfInvocationsInOrderChecker`
- [ ] #912 `Constructors`
- [x] #427 `FieldSetter`
- [x] #908 `FriendlyExceptionMaker`
- [x] #431 `HandyReturnValues` 
- [x] #432 `InvocationMarker`
- [x] #462 `InvocationsFinder`
- [x] #908 `JUnitDetecter`
- [x] #490 `MissingInvocationChecker`
- [x] #490 `MissingInvocationInOrderChecker`
- [x] #514 `MockUtil`
- [x] #503 `NonGreedyNumberOfInvocationsInOrderChecker`
- [x] #907 `NumberOfInvocationsInOrderChecker`
- [x] #907 `NumberOfInvocationsChecker`
- [x] #547 `ObjectMethodsGuru`
- [x] #427 `Reporter`
- [x] #535 `SuperTypesLastSorter`
- [x] #501 `TestMethodFinder`
- [x] #515 `VerifiableInvocationsFinder`

### PR DESCRIPTION
**Pull request by ChristianSchwarz**
relates to #426 

This PR simplifies the creation of ArgumentsAreDifferent-Exceptions. Saving 2 classes and ~40 lines of code.

### CODE REVIEW

File: src/main/java/org/mockito/internal/junit/JUnitTool.java [hunk 1]
```diff
@@ -4,15 +4,32 @@
...
 
-    private static final JUnitDetecter detecter = new JUnitDetecter();
+    private final static boolean hasJUnit = canCreateJUnitClass();
+
+    private JUnitTool() {
+    }
+
+    public static AssertionError createArgumentsAreDifferentException(String message, String wanted, String actual) {
+        if (hasJUnit()) {
+            return new org.mockito.exceptions.verification.junit.ArgumentsAreDifferent(message, wanted, actual);
```
Comment (TimvdLippe) [inline src/main/java/org/mockito/internal/junit/JUnitTool.java]: Can be the short name

Comment (TimvdLippe) [inline src/main/java/org/mockito/internal/junit/JUnitTool.java]: Hm oh nvm, I misread the package import

File: src/main/java/org/mockito/internal/junit/JUnitTool.java [hunk 2]
```diff
@@ -4,15 +4,32 @@
...
+        }
+        return new ArgumentsAreDifferent(message);
+    }
 
     public static boolean hasJUnit() {
```
Comment (TimvdLippe) [inline src/main/java/org/mockito/internal/junit/JUnitTool.java]: Is this method used somewhere else? Else we can just remove it and then on line 17 just remove the method invocation.

File: src/main/java/org/mockito/internal/junit/JUnitTool.java [hunk 3]
```diff
@@ -4,15 +4,32 @@
...
 
-    public static boolean hasJUnit() {
-        return detecter.hasJUnit();
+    /**     
+     * If JUnit is used,  AssertionError is returned that extends from ComparisonFailure      
+     * and hence provide a better IDE support as the comparison result is comparable      
+     */
+    public static AssertionError createArgumentsAreDifferentException(String message, String wanted, String actual) {
+        if (hasJUnit) {
+            return new org.mockito.exceptions.verification.junit.ArgumentsAreDifferent(message, wanted, actual);
```
Comment (szczepiq) [inline src/main/java/org/mockito/internal/junit/JUnitTool.java]: Can you test if this actually work in a project without JUnit? I want to make sure that the code does not throw class not found error or something like that. The previous code used reflection to construct the instance of the class that required JUnit on the classpath.

Perhaps we already validate this condition in our project (testng submodule) but I'm not sure right now. I will double check that over the weekend.

Comment (bric3) [inline src/main/java/org/mockito/internal/junit/JUnitTool.java]: It looks good to me. Yet better have a test with that. @ChristianSchwarz You can look at `ClassLoaders` util class that allows to create custom classloaders.

Comment (ChristianSchwarz) [inline src/main/java/org/mockito/internal/junit/JUnitTool.java]: Shame on me, it doesn't worked in an non JUnit env. !  Now it is fixed without the need of reflection. It was a good lesson for me about JVM class loading rules.

@bric3 thanks for pointing me to the `ClassLoaders` util!

**Comment by ChristianSchwarz** [comment]
@szczepiq @bric3 
The tests are in place now.

**Comment by ChristianSchwarz** [comment]
bump

**Comment by ChristianSchwarz** [comment]
Why the hell it is so important to end a java file with a new-line. Seriously why the build must break??? Same goes for a single whitespace at the end of a line. It adds a lot of more workload to fix this non issues, for no benefit.

**Comment by TimvdLippe** [comment]
@ChristianSchwarz The Checkstyle configuration is incubating at the moment. Primarily it enforces the same code style such that we as reviewers have an easier time and process pull requests faster (cutting out the noise). These issues you can check by running `./gradlew build` locally, so you should not rely on CI to error on them.

**Comment by ChristianSchwarz** [comment]
I know!

**Comment by ChristianSchwarz** [comment]
Better provide a formatter template, so it can be done by the IDE. Doing it by hand is a stubid frustrating job.

**Comment by TimvdLippe** [comment]
@ChristianSchwarz Hm I have configured my IntelliJ to do this automatically for me, so it is not bothering my workflow. Maybe you can configure your IDE to do the same? That was actually the goal of enforcing Checkstyle: every developer configures the IDE to fix this automatically.

**Comment by ChristianSchwarz** [comment]
Can you export your formatter and add a download-link to wiki contribution guidelines?

**Comment by ChristianSchwarz** [comment]
finally, ready to merge...

**Pull request merged**